transforms/helpers/eo-prop/private/method.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+  EOMethod,
+  EOPropertyForMethod,
+  FunctionExpression,
+  Identifier,
+} from '../../ast';
+import AbstractEOProp from './abstract';
+
+export default class EOMethodProp extends AbstractEOProp<EOPropertyForMethod> {
+  get value(): EOMethod | FunctionExpression {
+    return 'value' in this._prop ? this._prop.value : this._prop;
```

Comment:
Assuming you take my suggestion in `abstract.ts:18`

```suggestion
return this._prop.value ?? this._prop
```

Reply:
This broke type inference. But also, I think I am going to split these into separate classes since we're talking about allowing decorators on `EOMethod` by default but not on `FunctionExpression`.